File: engine/model/src/main/scala/net/server/channel/packet/movement/MoveSummonPacket.scala
package net.server.channel.packet.movement

import java.awt.Point

import net.server.MovementData

class MoveSummonPacket(private var _objectId: Int, private var _startPosition: Point, private var _hasMovement: Boolean,
                       private var _movementDataList: java.util.List[MovementData],
                       private var _movementList: java.util.List[java.lang.Byte]) extends BaseMovementPacket {
  def objectId: Int = _objectId

  def startPosition: Point = _startPosition

  def hasMovement: Boolean = _hasMovement

  def movementDataList: java.util.List[MovementData] = _movementDataList

  def movementList: java.util.List[java.lang.Byte] = _movementList

}

File: engine/engine-model/src/main/java/net/server/channel/packet/movement/MoveSummonPacket.java
package net.server.channel.packet.movement;

import java.awt.Point;
import java.util.List;

import net.server.MovementData;

public class MoveSummonPacket extends BaseMovementPacket {
   private final Integer objectId;

   private final Point startPosition;

   private final Boolean hasMovement;

   private final List<MovementData> movementDataList;

   private final List<Byte> movementList;

   public MoveSummonPacket(Integer objectId, Point startPosition, Boolean hasMovement, List<MovementData> movementDataList, List<Byte> movementList) {
      this.objectId = objectId;
      this.startPosition = startPosition;
      this.hasMovement = hasMovement;
      this.movementDataList = movementDataList;
      this.movementList = movementList;
   }

   public Integer objectId() {
      return objectId;
   }

   public Point startPosition() {
      return startPosition;
   }

   public Boolean hasMovement() {
      return hasMovement;
   }

   public List<MovementData> movementDataList() {
      return movementDataList;
   }

   public List<Byte> movementList() {
      return movementList;
   }
}
